packages/components/src/components/switch/switch-styles.ts

```diff
@@ -26,151 +26,140 @@ const getColors = (
   toggleBackgroundColorHover: string;
   textColor: string;
 } => {
-  const { backgroundColor, primaryColor, contrastHighColor, successColor, successColorDarken, disabledColor } =
-    getThemedColors(theme);
+  const { primaryColor, contrastMediumColor, successColor, successColorDarken, disabledColor } = getThemedColors(theme);
   const { backgroundColor: lightThemeBackgroundColor } = getThemedColors('light');
   const checkedColor = successColor;
   const disabledOrLoadingColor = isDisabledOrLoading && disabledColor;
 
   return {
-    backgroundColor,
-    buttonBorderColor: disabledOrLoadingColor || (checked ? checkedColor : contrastHighColor),
+    buttonBorderColor: disabledOrLoadingColor || (checked ? checkedColor : contrastMediumColor),
     buttonBorderColorHover: checked ? successColorDarken : primaryColor,
     buttonBackgroundColor: checked ? disabledOrLoadingColor || checkedColor : 'transparent',
     buttonBackgroundColorHover: checked ? successColorDarken : 'transparent',
     toggleBackgroundColor:
-      (!checked && disabledOrLoadingColor) || (checked ? lightThemeBackgroundColor : contrastHighColor),
+      (loading && 'transparent') ||
+      (disabled && !checked && disabledColor) ||
+      (checked ? lightThemeBackgroundColor : primaryColor),
     toggleBackgroundColorHover: checked ? lightThemeBackgroundColor : primaryColor,
     textColor: disabledOrLoadingColor || primaryColor,
   };
 };
 
-const getAlignLabelJssStyle: GetJssStyleFunction = (alignLabel: AlignLabel): JssStyle => {
-  const styles: Record<AlignLabel, JssStyle> = {
-    left: {
-      order: 0,
-      paddingLeft: 0,
-      paddingRight: spacingStaticSmall,
-    },
-    right: {
-      order: 1,
-      paddingLeft: spacingStaticSmall,
-      paddingRight: 0,
-    },
-  };
-  return styles[alignLabel];
-};
-
-const getStretchJssStyle: GetJssStyleFunction = (stretch: boolean): JssStyle => {
-  return stretch
-    ? {
-        width: '100%',
-        justifyContent: 'space-between',
-      }
-    : {
-        width: 'auto',
-        justifyContent: 'flex-start',
-      };
-};
-
 export const getComponentCss = (
   alignLabel: BreakpointCustomizable<AlignLabel>,
   hideLabel: BreakpointCustomizable<boolean>,
   stretch: BreakpointCustomizable<boolean>,
   checked: boolean,
+  disabled: boolean,
```

Comment:
There is some redundancy between `disabled` with  `loading` and `isDisabledOrLoading()`.